---
title: "WRC2015"
author: "R.B. Bledsoe and A.L. Peralta"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  pdf_document: default
  html_document:
    df_print: paged
question: How does long-term NPK deposition influence the diversity and composition
  of bulk soil microbiome and the plant rhizosphere microbiome of a C3 forb and a
  C4 grass?
description: Impact of long-term fertilization of bulk soils on plant rhizosphere
  micorbial communities.
---

```{r set working directory}
rm(list=ls())
#knitr::opts_knit$set(root.dir="~/GitHub/WRC15_Rhizo/")
#knitr::knit("~/GitHub/WRC15_Rhizo/WRC2015RhizoDec2018.Rmd")
#list.files()
```

```{r load packages and functions }
getwd()

#Set source R tools
source("../bin/DiversityFunctions.R")
source("../bin/MothurTools.R")

#load req'd packages
require("vegan")
require("dplyr")
require("tidyverse")
require("nlme")
require("reshape2")
require("ecodist")
require("ggplot2")
require("ade4")
require("png")
require("MASS")
require("grid")
require("ape")
require("png")
require("picante")
require("phyloseq")
require("plyr")
require("VennDiagram")
require("lme4")
require("multcomp")
require("agricolae")

#Set Std Err and Conf Int
se <- function(x, ...) {
  sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))
}
ci <- function(x, ...) {
  1.96 * sd(x, na.rm = TRUE)
}

set.seed(42)
```

```{r load data files and trim}
getwd()
#Load data files
#Assign file names to variables
sharedfile = "../data/wrc15rhizodenovo.17May2018.opti_mcc.unique_list.shared"
taxfile = "../data/wrc15rhizodenovo.17May2018.opti_mcc.unique_list.0.03.cons.taxonomy"
metafile = "../data/2015_WestResearchCampus_Sampling_Rhizo.csv"

#Read in design file
meta <- read.csv(metafile)
#Remove columns not needed for analysis (from example columns used for calculations)
meta<-meta[,-c(8:10,14:16,19:20)]
head(meta)
#Read in OTU file
otu <- read.otu(sharedfile)
#tax <- read.tax(taxfile)
#read.tax does not work and I was unable to solve issue. I use phyloseq import_mothur as a work around. 
tax <- import_mothur(mothur_constaxonomy_file =  taxfile)
tax.df <- as.data.frame(tax)
colnames(tax.df)=c("Domain","Phylum","Class", "Order","Family","Genus")
tax <- tax_table(tax.df)
tax

#Remove OTUs with less than 10 occurences across all sites
otu.2 <- otu[, which(colSums(otu) >9)]
#colnames(otu.2)
#rownames(otu.2)

#Remove all RP samples
#In the original sampling I collected bulk soils, and rhizospehre soils. Soil loosley adhering to roots was shaken into a bag (RP) for DNA and soil NH4/NO3 extraction. Roots and adhering soils collected for DNA processing (RS). Removed RP from analysis for clarity and bc very similar to RS communities.
otu.2 <-otu.2[!grepl("RP", rownames(otu.2)),]
#rownames(otu.2)
meta <- meta[!grepl("RP", meta$group),]

#What sample has lowest read count? WRC15AVRS4MF 43811
otu2<- colSums(t(otu.2))
#otu2
otu2[which.min(colSums(t(otu.2)))]
otu2[which.max(colSums(t(otu.2)))]

#graph of read counts, need to work on y axis
c<-as.data.frame(otu.2)
c$colsum <- colSums(t(c))
c$group <- rownames(c)
p<-ggplot(c, aes(x=group,y=colsum))+geom_bar(stat="identity")+coord_flip()
p

#Rarefy
#Set min sample number
min.N <- min(rowSums(otu.2))
#rrarefy returns a randomly rarefied community data frame or vector of selected size
#This is the otu set that should be used for beta diversity metrics
otu.rare <- rrarefy(otu.2, min.N)
```

```{r rarefaction curve}
#rarefy returns a species richness estimate in a random subsample of selected size
otu.rare.rich <- rarefy(x = otu.2, sample=min.N, se=T)
#Plot rarefaction curve
###I am not sure how to interpret this. I think it suggests that I did not capture all of the diversity? 
rarecurve(x = otu.2, step = 20, col="blue", cex=0.6, las=1)
abline(0,1,col='red')
text(1000,1000,"1:1",pos=2, col='red')

```
```{r Relative abundance and PERMANOVA }
#transforms read counts to relative abundance
#Using rarefied OTU dataset
otu.2.rel <- otu.2
for(i in 1:dim(otu.2)[1]){
  otu.2.rel[i,] <- otu.2[i,]/sum(otu.2[i,])
}

#Combine experimental deisng meta data with OTU relative abundance data
metaotu <- cbind(meta, otu.2.rel)
#Where does otu data begin, remove for PERMANOVA
#metaotu[1,20:21]

#PERMANOVA with plant and Treatment as factors
#Among all treatments plant (soil source) and treatment
metaotu.ad <- adonis(metaotu[,-c(1:19)] ~plant*Treatment, method = "bray", data = metaotu, perm=1000, set.seed=42)
metaotu.ad 

#Tries to make a 1GB vector and fails even with 1 permutation
#require("RVAideMemoire")
#metaotu.ad.1 <- metaotu[which(metaotu$Treatment=='M'),]
#p <- pairwise.perm.manova(metaotu.ad.1[,-c(1:19)],metaotu.ad.1$plant,nperm=1)

#Another attempt at pairwise adonis but not sure that this method has been well worked out by the authors
#require("pairwiseAdonis")
#data(metaotu)
#pairwise.adonis(metaotu[,-c(1:19)],metaotu$pxt)

#Plant (soil source) is significant but the PCoA suggests that it is only significant between plant and rhizosphere and not plant functionl type (grass, forb). Because I can't run a global pairwise analysis I will run individual pairwise analysis between plant(soil sources) within a treatment. 
#Plant (soil source) within treatment

#Pairwise adonis/PERMANOVA of soil source within a treatment
#MOWED UNFERTILIZED
metaotu.M <- metaotu[which(metaotu$Treatment=='M'),]
metaotu.M.GF <- metaotu.M[which(metaotu.M$plant!='bulk'),]
metaotu.M.BG <- metaotu.M[which(metaotu.M$plant!='ec'),]
metaotu.M.BF <- metaotu.M[which(metaotu.M$plant!='av'),]

metaotu.M.GF.ad <- adonis(metaotu.M.GF[,-c(1:19)] ~ plant, method = "bray", data = metaotu.M.GF, perm=1000, set.seed=42)
metaotu.M.GF.ad 

metaotu.M.BG.ad <- adonis(metaotu.M.BG[,-c(1:19)] ~ plant, method = "bray", data = metaotu.M.BG, perm=1000, set.seed=42)
metaotu.M.BG.ad 

metaotu.M.BF.ad <- adonis(metaotu.M.BF[,-c(1:19)] ~ plant, method = "bray", data = metaotu.M.BF, perm=1000, set.seed=42)
metaotu.M.BF.ad 

#MOWED FERTILIZED
metaotu.MF <- metaotu[which(metaotu$Treatment=='MF'),]
metaotu.MF.GF <- metaotu.MF[which(metaotu.MF$plant!='bulk'),]
metaotu.MF.BG <- metaotu.MF[which(metaotu.MF$plant!='ec'),]
metaotu.MF.BF <- metaotu.MF[which(metaotu.MF$plant!='av'),]

metaotu.MF.GF.ad <- adonis(metaotu.MF.GF[,-c(1:19)] ~ plant, method = "bray", data = metaotu.MF.GF, perm=1000, set.seed=42)
metaotu.MF.GF.ad 

metaotu.MF.BG.ad <- adonis(metaotu.MF.BG[,-c(1:19)] ~ plant, method = "bray", data = metaotu.MF.BG, perm=1000, set.seed=42)
metaotu.MF.BG.ad 

metaotu.MF.BF.ad <- adonis(metaotu.MF.BF[,-c(1:19)] ~ plant, method = "bray", data = metaotu.MF.BF, perm=1000, set.seed=42)
metaotu.MF.BF.ad 

```

```{r Calculate distance matrix and PCoA components }

otu.2.rel.dist<-vegdist(otu.2.rel, method="bray")

# Principal Coordinates Analysis
otu.2.rel.dist.pcoa <- cmdscale(otu.2.rel.dist, k=3, eig=TRUE, add=FALSE)
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1a <- round(otu.2.rel.dist.pcoa$eig[1] / sum(otu.2.rel.dist.pcoa$eig), 3) * 100
explainvar2a <- round(otu.2.rel.dist.pcoa$eig[2] / sum(otu.2.rel.dist.pcoa$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)

explainvar1a
explainvar2a

```

```{r PCoA Plot }
all.equal(rownames(meta), rownames(otu.2.rel))
# Set treatments
treat1 <- as.factor(meta$plant)
levels(treat1) <- ordered(c("bulk","av", "ec"))
treat2 <- as.factor(meta$Treatment)
levels(treat2) <- c("M", "MF")

pcoa.groups <- paste(meta$plant, meta$Treatment, sep = "_")

pcoa.points <- data.frame(otu.2.rel.dist.pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # Treatment
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # Plant

#pcoa.shape <- ordered(c("A. virginicus", "Bulk Soil", "E. caroliniana"))
pcoa.shape <- ordered(pcoa.shape,levels=c("bulk","av", "ec"))
pcoa.shape <- revalue(pcoa.shape, c("bulk"="Bulk Soil", "av"="Grass", "ec"="Forb"))
pcoa.shape <- mapvalues(pcoa.shape, from = c("bulk","av","ec"), to = c("Bulk Soil", "Grass", "Forb"))

levels(pcoa.col) <- c("Unfertilized","Fertilized")

#Removed RS samples becuase only have soil porperty values for RP samples
#Run envfit to see which soil varibales are predicators 
meta.env <- read.csv(metafile)
meta.env <- meta.env[!grepl("RS", meta.env$type),]
meta.env <- meta.env[,-c(1:11,13:16,18:20)]
fit<-envfit(otu.2.rel.dist.pcoa$points, meta.env, perm=1000, na.rm=T, set.seed=42)
fit

A <-as.list(fit$vectors)
vec<-as.data.frame(fit$vectors$arrows*sqrt(fit$vectors$r)*0.35)
p<- as.data.frame(A$pvals)
vec <- cbind(vec, p) 
vec <-  subset(vec, A$pvals<=0.05)

vec$parm<-rownames(vec)
colnames(vec)
#vec$parm<-c("Moisture","pH","C%","N%")

#plot PCoA
df1a <- as.data.frame(pcoa.cent.dataframe)
plot1 <- ggplot(df1a, aes(x=V1, y=V2,  color=pcoa.col, shape = pcoa.shape, 
                 group = interaction(pcoa.col, pcoa.shape))) + theme_bw() 

plot1 + geom_point(aes(shape = pcoa.shape, colour = pcoa.col), size=7) + 
  #removes gridlines from plot
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  #Plot soil property vectors from envfit
  geom_segment(data=vec, aes(x=0,xend=Dim1,y=0,yend=Dim2), size=1, 
      arrow = arrow(length = unit(0.2, "cm")),colour="black",inherit.aes=F)+ 
  #Label soil vectors. Using annotate because I can't figure out how to move segment text
  #Note when re-run should double check envfit stats to make sure that soil parameters are the same
  #geom_text(data=vec,aes(x=Dim1,y=Dim2,label=vec$parm),size=5, inherit.aes=F)+
  annotate("text",x=-0.02,y=0.2, label="C%") +
  annotate("text",x=0.02,y=0.2, label="N%") +
  annotate("text",x=0,y=-.2, label="moisture") +
  annotate("text",x=0.1,y=-.225, label="pH") +
  #Set error bacrs for geom_point
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.02), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.02), colour="black") + 
  #Set colors for treatments
  scale_colour_manual(values = c("#999999", "#000000")) +
  #Would have liked to set shapes but couldn't get this to work
  #scale_shape_manual(values = c(22, 21, 24)) +
  #Sets map coordinates
  coord_cartesian(xlim = c(-0.3, 0.3), ylim = c(-0.3, 0.3)) +
  #Sets axis text and put border around plot
  theme(axis.title = element_text(size=14), 
        axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
        panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=14)) +
  #Set legend text size
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  #Sets size of tick marks on axis
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  #Sets labels for plot title, axis titles, and legend headings
  xlab("PCoA 1 (31.6%)") + ylab("PCoA 2 (22.5%)") + 
  labs(shape = "Soil Source") +
  labs(colour="Treatment") +
  ggtitle("16S rRNA Microbial Community Analysis") 
  
#Save a copy of the plot
ggsave("../figures/16SrRNA_WRC2015Rhizo.jpg", plot=last_plot(), device=NULL, path=NULL, scale=1, width=8.7, height=6.3, dpi=900, limitsize=TRUE)

```

```{r Presence_Absence and Species richness }
#Transform data to reflect only presence or absence ie 0 or 1
otu.2.PA<-(otu.rare>0*1)
#sanity check 
#otu.2.PA[,1:5]
#Add meta data
metaotu.2 <- cbind(meta, otu.2.PA)
#metaotu.2[1:15]
#Transform True/False to 0/1 
cols <- sapply(metaotu.2, is.logical)
metaotu.2[,cols]<-lapply(metaotu.2[,cols],as.numeric)
#head(metaotu.2)
```


```{r diversity metrics: richness}
#chao1 richness index
chao1 <- estimateR(otu.2)
c <- as.data.frame(chao1)
ct <- t(c)
otu.div <- cbind(meta, ct)

#interaction variable so can use Tukey
i <- with(otu.div, interaction(Treatment, plant))

richness.lm <- aov(S.chao1 ~ Treatment + plant + i, data = otu.div)
plot(richness.lm)
anova(richness.lm)
#Get Tukey post hoc groups p<=0.01*
t <- HSD.test(richness.lm, "i", group=TRUE, alpha=.01)
t
plot(t)

#Set variable names for plot
levels(meta$Treatment) <- c("Unfertilized", "Fertilized")
p <- meta$plant
p <- ordered(p,levels=c("bulk","av", "ec"))
p <- revalue(p, c("bulk"="Bulk Soil", "av"="Grass", "ec"="Forb"))
p <- mapvalues(p, from = c("bulk","av","ec"), to = c("Bulk Soil", "Grass", "Forb"))

#Plot
chao<- ggplot(otu.div, aes(x=p, y=S.chao1, fill=Treatment))  +
  geom_boxplot(binaxis='y', stackdir='center') + scale_fill_manual(values=c("#999999", "#000000")) +
#Set labels for Tukey groups
annotate("text",x=0.82,y=5200, label="b") +
annotate("text",x=1.81,y=5400, label="b") +
annotate("text",x=2.81,y=5725, label="b") +
annotate("text",x=1.1815,y=7050, label="a") +
annotate("text",x=2.19,y=6600, label="bc") + 
annotate("text",x=3.19,y=6250, label="ab") + 
#Set plot elements
theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_blank(), 
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          labs(x = "", y = "Chao1 species richness")
chao
#Save a copy of the plot
ggsave("../figures/chao1.jpg", plot=chao, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)


```

```{r diversity metrics - calculate diversity and evenness}
# Calculate diversity metrics
shannon <- diversity(otu.rare, "shannon")
simp <- diversity(otu.rare, "simpson")
invsimp <-diversity(otu.rare, "invsimpson")
# Pielous evenness (uses presence/absence data)
J <- shannon/log(specnumber(otu.2.PA[,-c(1:1)]))

#Adding diversity fields to dataset
otu.div$shannon <- NULL
otu.div$J <- NULL
otu.div$simp <- NULL
otu.div$invsimp <- NULL
otu.div <- cbind(otu.div,shannon,J, simp, invsimp)
```

```{r diversity metrics - display shannon }
#Run richness first. interaction, levels, and treatments for plotting set there
shannon.lm <- aov(shannon ~ Treatment + plant + i, data = otu.div)
plot(shannon.lm)
anova(shannon.lm)

#Tukey groups at p<=0.01*
t <- HSD.test(shannon.lm, "i", group=TRUE, alpha=0.01)
t
plot(t)

#p is set in richness chuck
shan<- ggplot(otu.div, aes(x=p, y=shannon, fill=Treatment))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000"))  + annotate("text",x=0.82,y=6.1, label="bc") +
annotate("text",x=1.81,y=5.9, label="c") +
annotate("text",x=2.81,y=6.05, label="c") +
annotate("text",x=1.1815,y=6.5, label="a") +
annotate("text",x=2.19,y=6.3, label="ab") + 
annotate("text",x=3.19,y=6.35, label="ab") + 
#Set plot elements
theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_blank(), 
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          labs(x = "", y = "Shannon diversity H'")

shan

ggsave("../figures/shannon.jpg", plot=shan, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)
```
```{r diversity metrics - display simp and invsimp}

simp.lm <- aov(simp ~ Treatment + plant + i, data = otu.div)
plot(simp.lm)
anova(simp.lm)
#TukeyHSD(simp.lm)
t <- HSD.test(simp.lm, "i", group=TRUE, alpha=0.01)
t
plot(t)

invsimp.lm <- aov(invsimp ~ Treatment + plant + i, data = otu.div)
plot(invsimp.lm)
anova(invsimp.lm)
#TukeyHSD(invsimp.lm)
t <- HSD.test(invsimp.lm, "i", group=TRUE, alpha=0.01)
t
plot(t)

isimp<- ggplot(otu.div, aes(x=p, y=invsimp, fill=Treatment))  +
  geom_boxplot(binaxis='y', stackdir='center') + scale_fill_manual(values=c("#999999", "#000000"))
#Groups annotated by alpha=0.05 on plot --- need to correst to alpha=0.01 
isimp + annotate("text",x=0.81,y=105, label="bc") +
 annotate("text",x=1.81,y=70, label="c") +
 annotate("text",x=2.81,y=95, label="c") +
 annotate("text",x=1.1815,y=155, label="a") +
 annotate("text",x=2.19,y=110, label="bc") +
   annotate("text",x=3.19,y=167, label="ab") +
  theme_bw()  +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=18,face="bold"),
          axis.text=element_text(size=18), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Soil Source", y = "Inverse Simpson diversity index") + 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1)) +
   theme(legend.text=element_text(size=14), legend.title = element_text(size=14))
```

```{r diversity metrics - Pielous evennes J}
J.lm <- aov(J ~ Treatment + plant + i , data = otu.div)
plot(J.lm)
anova(J.lm)
#TukeyHSD(J.lm)
t <- HSD.test(J.lm, "i", group=TRUE, alpha=0.01)
t
plot(t)

even<- ggplot(otu.div, aes(x=p, y=J, fill=Treatment))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000")) +
#Set Tukey groups
annotate("text",x=0.82,y=0.77, label="ab") +
annotate("text",x=1.81,y=0.75, label="b") +
annotate("text",x=2.81,y=0.76, label="b") +
annotate("text",x=1.1815,y=0.80, label="a") +
annotate("text",x=2.19,y=0.78, label="a")  + 
annotate("text",x=3.19,y=0.80, label="a")  + 
#Set plot properties
theme_bw()  +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
          axis.title.y=element_text(margin=margin(r=10)), 
          axis.title.x=element_text(margin=margin(t=10))) +
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(size=14), panel.border = 
          element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Soil Source", y = "Pileou's eveness") + 
    theme(legend.text=element_text(size=14), legend.title = element_text(size=14), 
          legend.position = "bottom")

even 

ggsave("../figures/even.jpg", plot=even, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

```

```{r gene copy analysis}
#These are the .txt files generated from running RDP classifier to estimate gene copy number
#To keep with rest of analysis I removed RP samples
filename <- c("2M", "2MF", "4M","4MF","6M","6MF","8M","8MF","AVRS2M", "AVRS2MF", "AVRS4M","AVRS4MF","AVRS6M","AVRS6MF","AVRS8M","AVRS8MF","ECRS2M", "ECRS2MF", "ECRS4M","ECRS4MF","ECRS6M","ECRS6MF","ECRS8M","ECRS8MF")

#full list
#filename <- c("2M", "2MF", "4M","4MF","6M","6MF","8M","8MF","AVRP2M", "AVRP2MF", "AVRP4M","AVRP4MF","AVRP6M","AVRP6MF","AVRP8M","AVRP8MF","AVRS2M", "AVRS2MF", "AVRS4M","AVRS4MF","AVRS6M","AVRS6MF","AVRS8M","AVRS8MF","ECRP2M", "ECRP2MF", "ECRP4M","ECRP4MF","ECRP6M","ECRP6MF","ECRP8M","ECRP8MF","ECRS2M", "ECRS2MF", "ECRS4M","ECRS4MF","ECRS6M","ECRS6MF","ECRS8M","ECRS8MF")

#Use meta - already has RP removed, add gene copy number columns
new_col <- c("gc_adjusted","otu_count","bac_gnc","copio_otu","copio_class","oligo_otu","oligo_class", "class_ratio","otu_ratio","copio_phylum", "oligo_phylum","phylum_ratio","copio_order", "oligo_order","order_ratio","copio_family", "oligo_family","family_ratio")
meta[new_col] <- 0

#24 is the number of filenames
for (i in 1:24) {
#Each file represents a sample, for each sample get gene copy number by dividing adjusted relative  abundance by otu_count for that group to get gene copy number. Get gene copy number by OTU and by class, bin as copio or oligo, sum number of oligo and copio per file/sample, and then get a copio:oligo ratio

#set file path and file name. Tried using ~/GitHub/... but didn't work   
soilfile <- paste("../data/gene_copy/", filename[i] , "_soil_hier.txt", sep="")
adjustedfile <-  paste("../data/gene_copy/cnadjusted_", filename[i], "_soil_hier.txt", sep="")

#read in files
#Soilheir has the otu count for each taxonomic grouping
#cnadjusted files has the adjusted otu count
#adjusted in that RDP calculates the adjusted otu_count from the original otu_count/gene copy number of that group. This gives the otu_count with consideration thatthe number of reads for an organism could be inflated if they have many rrn gene copies within an individual.  
soilheir <- read.delim(soilfile,  header=T)
adjusted <- read.delim(adjustedfile, header=T)
#Renames count columns something more friendly
colnames(soilheir)[5] <- "otu_count"
colnames(adjusted)[5] <- "adjusted"

#Add otu_count, adjusted count, and calculated gene copy number into one dataset
soil.adjusted <- cbind(soilheir, adjusted= adjusted$adjusted)
soil.adjusted <- soil.adjusted %>% mutate(gnc = otu_count/adjusted)

#Filter by taxonomic rank
soil.phylum <- soil.adjusted %>% filter(rank=="phylum")
soil.class <- soil.adjusted %>% filter(rank=="class")
soil.order <- soil.adjusted %>% filter(rank=="order")
soil.family <-soil.adjusted %>% filter(rank=="family")
soil.genus <- soil.adjusted %>% filter(rank=="genus")  

#Using a filtered dataset from above count the number of rows that correspond to copio or oligo
#If an otu has a gene count greater than 5 it is considered copiotroph
#If an otu has a gene count less than 5 then it is considered oligotroph

#By Phylum
soil.p.c <- soil.phylum %>% 
  filter(gnc >= 5) 
  c <- nrow(soil.p.c)
  
soil.p.o <- soil.phylum %>% 
  filter(gnc < 5) 
  o <- nrow(soil.p.o)
  
c_to_o <- c/o

#Add to copio, oligo, and ratio to meta table 
meta <- meta %>% 
  mutate(copio_phylum = ifelse(group == paste("WRC15", filename[i], sep="") , c, copio_phylum)) %>%   
  mutate(oligo_phylum = ifelse(group == paste("WRC15", filename[i], sep="") , o, oligo_phylum)) %>%   
  mutate(phylum_ratio = ifelse(group == paste("WRC15", filename[i], sep="") , c_to_o, phylum_ratio))

#By Class
soil.class.c <- soil.class %>% 
  filter(gnc >= 5) 
  c <- nrow(soil.class.c)
  
soil.class.o <- soil.class %>% 
  filter(gnc < 5) 
  o <- nrow(soil.class.o)
  
c_to_o <- c/o

#Add to copio, oligo, and ratio to meta table 
meta <- meta %>% 
  mutate(copio_class = ifelse(group == paste("WRC15", filename[i], sep="") , c, copio_class)) %>%   
  mutate(oligo_class = ifelse(group == paste("WRC15", filename[i], sep="") , o, oligo_class)) %>%   
  mutate(class_ratio = ifelse(group == paste("WRC15", filename[i], sep="") , c_to_o, class_ratio))

#By Order
soil.o.c <- soil.order %>% 
  filter(gnc >= 5) 
  c <- nrow(soil.o.c)
  
soil.o.o <- soil.order %>% 
  filter(gnc < 5) 
  o <- nrow(soil.o.o)
  
c_to_o <- c/o

#Add to copio, oligo, and ratio to meta table 
meta <- meta %>% 
  mutate(copio_order = ifelse(group == paste("WRC15", filename[i], sep="") , c, copio_order)) %>%   
  mutate(oligo_order = ifelse(group == paste("WRC15", filename[i], sep="") , o, oligo_order)) %>%   
  mutate(order_ratio = ifelse(group == paste("WRC15", filename[i], sep="") , c_to_o, order_ratio))
#By Order
soil.o.c <- soil.order %>% 
  filter(gnc >= 5) 
  c <- nrow(soil.o.c)
  
soil.o.o <- soil.order %>% 
  filter(gnc < 5) 
  o <- nrow(soil.o.o)
  
c_to_o <- c/o

#Add to copio, oligo, and ratio to meta table 
meta <- meta %>% 
  mutate(copio_order = ifelse(group == paste("WRC15", filename[i], sep="") , c, copio_order)) %>%   
  mutate(oligo_order = ifelse(group == paste("WRC15", filename[i], sep="") , o, oligo_order)) %>%   
  mutate(order_ratio = ifelse(group == paste("WRC15", filename[i], sep="") , c_to_o, order_ratio))

#By Family
soil.f.c <- soil.family %>% 
  filter(gnc >= 5) 
  c <- nrow(soil.f.c)
  
soil.f.o <- soil.family %>% 
  filter(gnc < 5) 
  o <- nrow(soil.f.o)
  
c_to_o <- c/o

#Add to copio, oligo, and ratio to meta table 
meta <- meta %>% 
  mutate(copio_family = ifelse(group == paste("WRC15", filename[i], sep="") , c, copio_family)) %>%   
  mutate(oligo_family = ifelse(group == paste("WRC15", filename[i], sep="") , o, oligo_family)) %>%   
  mutate(family_ratio = ifelse(group == paste("WRC15", filename[i], sep="") , c_to_o, family_ratio))
#By Order
soil.o.c <- soil.order %>% 
  filter(gnc >= 5) 
  c <- nrow(soil.o.c)
  
soil.o.o <- soil.order %>% 
  filter(gnc < 5) 
  o <- nrow(soil.o.o)
  
c_to_o <- c/o

#Add to copio, oligo, and ratio to meta table 
meta <- meta %>% 
  mutate(copio_order = ifelse(group == paste("WRC15", filename[i], sep="") , c, copio_order)) %>%   
  mutate(oligo_order = ifelse(group == paste("WRC15", filename[i], sep="") , o, oligo_order)) %>%   
  mutate(order_ratio = ifelse(group == paste("WRC15", filename[i], sep="") , c_to_o, order_ratio))

#By OTU
soil.otu.c <- soil.genus %>% 
  filter(gnc >= 5) 
  c <- nrow(soil.otu.c)
  
soil.otu.o <- soil.genus %>% 
  filter(gnc < 5) 
  o <- nrow(soil.otu.o)
  
c_to_o <- c/o

meta <- meta %>% 
  mutate(copio_otu = ifelse(group == paste("WRC15", filename[i], sep="") , c, copio_otu)) %>%   
  mutate(oligo_otu = ifelse(group == paste("WRC15", filename[i], sep="") , o, oligo_otu)) %>%       
  mutate(otu_ratio = ifelse(group == paste("WRC15", filename[i], sep="") , c_to_o, otu_ratio))

#remove datasets generated to be tidy
#rm(soilheir, adjusted, soil.adjusted, soil.domain, soil.phylum, soil.class, soil.genus, soil.order, #soil.class.c, soil.class.o, soil.otu.c, soil.otu.o)

}

```

```{r visual gene copy number and stats}

levels(meta$Treatment) <- c("Unfertilized", "Fertilized")

p <- meta$plant
p <- ordered(p,levels=c("bulk","av", "ec"))
p <- revalue(p, c("bulk"="Bulk Soil", "av"="Grass", "ec"="Forb"))
p <- mapvalues(p, from = c("bulk","av","ec"), to = c("Bulk Soil", "Grass", "Forb"))

gene<- ggplot(meta, aes(x=p, y=otu_ratio, fill=Treatment))  +
  geom_boxplot()+
  scale_fill_manual(values=c("#999999", "#000000")) +
  #Tukey groups
  annotate("text",x=0.8,y=0.24, label="a") +
  annotate("text",x=1.81,y=0.16, label="b") +
  annotate("text",x=2.8,y=0.18, label="ab") +
  annotate("text",x=1.2,y=0.19, label="ab") +
  annotate("text",x=2.2,y=0.18, label="ab") +
  annotate("text",x=3.2,y=0.18, label="ab") +
  #Clear gridlines from plot and set plot border
  theme_bw()  +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
          axis.title.y=element_text(margin=margin(r=10)), 
          axis.title.x=element_text(margin=margin(t=10))) +
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(size=14), panel.border = 
          element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Soil Source", y = "Copitroph:Oligotroph") + 
    theme(legend.text=element_text(size=14), legend.title = element_text(size=14), 
          legend.position = "bottom")

gene

ggsave("../figures/gene_copy.jpg", plot=gene, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

#Set interaction, run anova, and then Tukey if plant significant
i <- with(meta, interaction(Treatment, plant))
gnc <- aov(otu_ratio ~ Treatment + plant + i , data=meta)
plot(gnc)
anova(gnc)
#Tukey groups
t <- HSD.test(gnc, "i", group=TRUE, alpha=0.01)
t
plot(t)

```



```{r environmental parameters}
meta.env <- read.csv(metafile)
#Comment below line if want to run soil properties for entire experiemnt
#Run below line if want to look at bulk and RS only
meta.env <- meta.env[!grepl("RS", meta.env$type),]

#NEED A SUMMARY TABLE
by_plant_treatment <- meta.env %>%
  group_by(plant, Treatment) %>%
  summarise_all(funs(mean,sd), na.rm=T)

by_treatment <- meta.env %>%
  group_by(Treatment) %>%
  summarise_all(funs(mean,sd), na.rm=T)

#i <- with(meta.env, interaction(Treatment, plant))

nh4 <- aov(nh4_mg.l ~ Treatment, meta.env)
plot(nh4)
anova(nh4)

nh4<- ggplot(meta.env, aes(x=Treatment, y=nh4_mg.l, fill=plant))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000","333333"))
nh4


pH <- aov(pH ~ Treatment , meta.env)
plot(pH)
anova(pH)

ppH<- ggplot(meta.env, aes(x=Treatment, y=pH, fill=plant))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000", "333333"))
ppH

#t <- HSD.test(pH, "i" , group=TRUE, alpha=0.05)
#t
#plot(t)

no3 <- aov(no3_mg.l ~ Treatment , data=meta.env) 
plot(no3)
anova(no3)

avg_temp_C <- aov(avg_temp_C ~ Treatment , data=meta.env) 
plot(avg_temp_C)
anova(avg_temp_C)

moist_percent <- aov(moist_percent ~ Treatment , data=meta.env) 
plot(moist_percent)
anova(moist_percent)

TN <- aov(TN ~ Treatment , data=meta.env) 
plot(TN)
anova(TN)

c_percent <- aov(c_percent ~ Treatment , data=meta.env) 
plot(c_percent)
anova(c_percent)

n_percent <- aov(n_percent ~ Treatment , data=meta.env) 
plot(n_percent)
anova(n_percent)

C_N <- aov(C_N ~ Treatment , data=meta.env) 
plot(C_N)
anova(C_N)

```
```{r does shannon diversity correlate with gene copy # or pH env vars}
#Yes it appears so..
all <- cbind(meta, otu.div$shannon)
#all <- all[!grepl("RS", all$type),]

gc <- ggplot(all, aes(x=meta.env$pH, y=shannon, color=Treatment))  +
  geom_point()+ scale_color_manual(values=c("#999999", "#000000")) +
  geom_smooth(method=lm, se=T)+
  theme_bw()  +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
          axis.title.y=element_text(margin=margin(r=10)), 
          axis.title.x=element_text(margin=margin(t=10))) +
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(size=14), panel.border =
          element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "soil pH", y = "Shannon diversity H'") + 
    theme(legend.text=element_text(size=14), legend.title = element_text(size=14), 
          legend.position = "bottom")

gc

ggsave("../figures/shannon_pH.jpg", plot=gc, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

gc1 <- ggplot(all, aes(x=otu_ratio, y=shannon, color=Treatment))  +
  geom_point()+ scale_color_manual(values=c("#999999", "#000000")) +
  geom_smooth(method=lm, se=T) +
  theme_bw()  +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
          axis.title.y=element_text(margin=margin(r=10)), 
          axis.title.x=element_text(margin=margin(t=10))) +
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(size=14), panel.border =
          element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Copiptroph:Oligotroph", y = "Shannon diversity H'") + 
    theme(legend.text=element_text(size=14), legend.title = element_text(size=14), 
          legend.position = "bottom")
gc1

ggsave("../figures/shannon_otu.jpg", plot=gc1, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

gcfit <- lm(shannon ~ otu_ratio  , data=meta)
anova(gcfit)
summary(gcfit)

gcfit1 <- lm(shannon ~ pH, data=meta.env)
anova(gcfit1)
summary(gcfit1)

#I think this is saying that 52% of the variation in diversity is due to pH. But none to little 
#On graph there is also a clear separation between fertilzaiton treatments
```

```{r load taxonomy data into phyloseq object}
# Import otu and tax into phyloseq
otu <- otu_table(t(otu.2), taxa_are_rows = T)
tax <- import_mothur(mothur_constaxonomy_file =  taxfile)
#rownames(meta) <- meta$group
sample <- sample_data(meta)
#import into phyloseq
phy<-phyloseq(otu,tax)
row.names(sample)=meta$group
sample_names(sample)
#Assign rownames to be Sample ID's
# Merge mothurdata object with sample metadata
meta_merge <- merge_phyloseq(phy,sample)

#meta_merge

#assign column names
colnames(tax_table(meta_merge))
colnames(tax_table(meta_merge)) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus")
colnames(tax_table(meta_merge))
rank_names(meta_merge)

#Inspecting phyloseq object
#sample_variables(meta_merge)
#sample_names(meta_merge)
#tax_table(meta_merge)[1:5, 2:6]

treat_merge = merge_samples(meta_merge, sample_data(meta_merge)$pxt)
otu_table(treat_merge)[1:5,1:5]
```

```{r phyloseq Phylum stacked graphs of taxa composition}

# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample
wrc_phylum <- treat_merge %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.01) %>%                        # Filter out low abundance taxa
  arrange(Phylum)                                      # Sort data frame alphabetically by phylum

# Set colors for plotting
phylum_colors <- c("brown","orange", "blue", "red","yellow","purple")
  
  #c("#CBD588", "#5F7FC7","#DA5724", "#508578", "#CD9BCD",
  # "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
  #"#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861", "#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

# Plot 
s <- as.factor(wrc_phylum$Sample)
wrc_phylum$Sample <- ordered(s,levels=c("bulkM","avM", "ecM","bulkMF","avMF", "ecMF"))
s <- revalue(s, c("bulkM"="bulk unfertilized", "avM"="grass unfertilized", "ecM"="forb unfertilized","bulkMF"="bulk fertilized", "avMF"="grass fertilized", "ecMF"="forb fertilized"))
s <- mapvalues(s, from = c("bulkM","avM", "ecM","bulkMF","avMF", "ecMF"), to = c("bulk unfertilized","grass unfertilized", "forb unfertilized","bulk fertilized", "grass fertilized", "forb fertilized"))

wrc_plot<- ggplot(wrc_phylum, aes(x = Phylum, y = Abundance, fill = s)) + 
  #facet_grid(Type ~ .) +
  #####UPDATE THIS ON EACH GRAPH ######
  geom_bar(stat = "summary", fun.y = "mean") +
  scale_fill_manual(values = phylum_colors) +
  # Remove x axis title
  theme(axis.title.x = element_blank()) + 
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Phyla > 1.0%) \n") +
  ggtitle("Phylum Composition of Bacterial Communities By Site") 

wrc_plot + coord_flip()

ggsave(file="../figures/phylum_stack.pdf", width=9, height=9, dpi=300)

```

```{r phyloseq Class stacked graphs of taxa composition}
# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample
wrc_class <- treat_merge %>%
  tax_glom(taxrank = "Class") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.01) %>%                        # Filter out low abundance taxa
  arrange(Class)                                     # Sort data frame alphabetically by phylum

wrc_class
# Set colors for plotting
class_colors <- c("brown","orange", "blue", "red","yellow","purple")
  #c(
  #"#CBD588", "#5F7FC7","#DA5724", "#508578", "#CD9BCD",
  # "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
  #"#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861", "#000000", "#E69F00", "#56B4E9", #"#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7", "green", "red", "yellow","tan", #"pink", "purple", "gray", "white", "339900","666699","996699","009966")

# Plot 

s <- as.factor(wrc_class$Sample)
#wrc_class$Sample <- ordered(s,levels=c("bulkM","avM", "ecM","bulkMF","avMF", "ecMF"))
#t <- as.factor(wrc_class$plant)
#wrc_class$plant <- ordered(t, levels=c("1","2","3"))
#u <- as.factor(wrc_class$Treatment)
#wrc_class$Treatment <- ordered(u, levels=c("1","2"))

t <- revalue(s, c("bulkM"="bulk unfertilized", "avM"="grass unfertilized", "ecM"="forb unfertilized","bulkMF"="bulk fertilized", "avMF"="grass fertilized", "ecMF"="forb fertilized"))
t <- mapvalues(s, from = c("bulkM","avM", "ecM","bulkMF","avMF", "ecMF"), to = c("bulk unfertilized","grass unfertilized", "forb unfertilized","bulk fertilized", "grass fertilized", "forb fertilized"))
t<-ordered(t,levels=c("bulk unfertilized","grass unfertilized", "forb unfertilized","bulk fertilized", "grass fertilized", "forb fertilized"))


wrc_plot<- ggplot(wrc_class, aes(x = Class, y = Abundance, fill = t)) + 
  #facet_grid(plant ~ .) + 
  geom_bar(stat = "identity", width=0.5) + #scale_fill_manual(values=c("#111111","#777777"))+
  scale_fill_manual(values = class_colors) +
  # Remove x axis title
 theme_bw()  +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
          axis.title.y=element_text(margin=margin(r=10)), 
          axis.title.x=element_text(margin=margin(t=10)), panel.border =
          element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + 
  #theme(axis.title=element_text(vjust=1,size=14,face="bold"),
  #        axis.text=element_text(size=14), axis.text.x = element_text(size=14))+
  ylab("Class Relative Abundance > 1.0%") +
  xlab("Treatment") +
  labs(fill="Treatment") +
  ggtitle("Class Composition of Bacterial Communities") 

wrc_plot + coord_flip() 

ggsave(file="../figures/class_stack.pdf", width=9, height=9, dpi=300)

wrc_class_o<-wrc_class[,c(2:3,23:25)]

write.csv(wrc_class_o, file="class_composition.csv", row.names=TRUE)
```
```{r phyloseq Family - Actinobacteria  stacked graphs of taxa composition}
# melt to long format (for ggploting) 
# prune out phyla below 1% in each sample
wrc_family <- treat_merge %>%
  tax_glom(taxrank = "Family") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.01, Class == "Actinobacteria") %>%                        # Filter out low abundance taxa
  arrange(Family)                                     # Sort data frame alphabetically by phylum

# wrc_family
# Set colors for plotting
family_colors <- c("brown","orange", "blue", "red","yellow","purple")
  #"#CBD588", "#5F7FC7","#DA5724", "#508578", "#CD9BCD",
  # "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
 #"#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861", "#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7", "green", "red", "yellow","tan", "pink", "purple", "gray", "white", "339900","666699","996699","black"
#)

# Plot 

s <- as.factor(wrc_family$Sample)
t <- revalue(s, c("bulkM"="bulk unfertilized", "avM"="grass unfertilized", "ecM"="forb unfertilized","bulkMF"="bulk fertilized", "avMF"="grass fertilized", "ecMF"="forb fertilized"))
t <- mapvalues(s, from = c("bulkM","avM", "ecM","bulkMF","avMF", "ecMF"), to = c("bulk unfertilized","grass unfertilized", "forb unfertilized","bulk fertilized", "grass fertilized", "forb fertilized"))
t<-ordered(t,levels=c("bulk unfertilized","grass unfertilized", "forb unfertilized","bulk fertilized", "grass fertilized", "forb fertilized"))

wrc_plot<- ggplot(wrc_family, aes(x = Family, y = Abundance, fill = t)) + 
  #facet_grid(Type ~ .) +
  geom_bar(stat = "identity", width=0.5) +
  scale_fill_manual(values = family_colors) +
  # Remove x axis title
  theme_bw()  +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
          axis.title.y=element_text(margin=margin(r=10)), 
          axis.title.x=element_text(margin=margin(t=10)), panel.border =
          element_rect(colour = "black",size=1), legend.position = "none") + 
    theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(size=14))+
  ylab("Relative Abundance > 1%") +
  xlab("Family Name")+
  #labs(fill="Treatment")+
  ggtitle("Actinobacteria Composition of \n Bacterial Communities") 

wrc_plot + coord_flip() 

wrc_family_actino<-wrc_family[,c(2:3,23:27)]

write.csv(wrc_family_actino, file="actino_family_composition.csv", row.names=TRUE)

ggsave(file="../figures/actino_family_stack.pdf", width=15, height=9, dpi=300)

```

```{r phyloseq Family - Alphaproteobacteria  stacked graphs of taxa composition}
# melt to long format (for ggploting) 
# prune out phyla below 1% in each sample
wrc_family <- treat_merge %>%
  tax_glom(taxrank = "Family") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.005, Class == "Alphaproteobacteria") %>%                        # Filter out low abundance taxa
  arrange(Family)                                     # Sort data frame alphabetically by phylum

# wrc_family
# Set colors for plotting
family_colors <- c("brown","orange", "blue", "red","yellow","purple")
  #"#CBD588", "#5F7FC7","#DA5724", "#508578", "#CD9BCD",
  # "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
 #"#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861", "#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7", "green", "red", "yellow","tan", "pink", "purple", "gray", "white", "339900","666699","996699","black"
#)

# Plot 

s <- as.factor(wrc_family$Sample)
t <- revalue(s, c("bulkM"="bulk unfertilized", "avM"="grass unfertilized", "ecM"="forb unfertilized","bulkMF"="bulk fertilized", "avMF"="grass fertilized", "ecMF"="forb fertilized"))
t <- mapvalues(s, from = c("bulkM","avM", "ecM","bulkMF","avMF", "ecMF"), to = c("bulk unfertilized","grass unfertilized", "forb unfertilized","bulk fertilized", "grass fertilized", "forb fertilized"))
t<-ordered(t,levels=c("bulk unfertilized","grass unfertilized", "forb unfertilized","bulk fertilized", "grass fertilized", "forb fertilized"))

wrc_plot<- ggplot(wrc_family, aes(x = Family, y = Abundance, fill = t)) + 
  #facet_grid(Type ~ .) +
  geom_bar(stat = "identity", width=0.5) +
  scale_fill_manual(values = family_colors) +
  # Remove x axis title
  theme_bw()  +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
          axis.title.y=element_text(margin=margin(r=10)), 
          axis.title.x=element_text(margin=margin(t=10)), panel.border =
          element_rect(colour = "black",size=1), legend.position = "none") + 
    theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(size=14))+
  ylab("Relative Abundance > 1%") +
  xlab("Family Name")+
  #labs(fill="Treatment")+
  ggtitle("Alphaproteobacteria Composition of \n Bacterial Communities") 

wrc_plot + coord_flip() 

wrc_family_alpha<-wrc_family[,c(2:3,40:45)]

write.csv(wrc_family_alpha, file="alpha_family_composition.csv", row.names=TRUE)

ggsave(file="../figures/alpha_family_stack.pdf", width=15, height=9, dpi=300)

```

```{r phyloseq order stacked graphs of taxa composition}
# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample
wrc_order <- treat_merge %>%
  tax_glom(taxrank = "Order") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.01) %>%                        # Filter out low abundance taxa
  arrange(Order)                                   # Sort data frame alphabetically by phylum

# wrc_family
# Set colors for plotting
family_colors <- c("brown","orange", "blue", "red","yellow","purple")
  #"#CBD588", "#5F7FC7","#DA5724", "#508578", "#CD9BCD",
  # "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
 #"#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861", "#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7", "green", "red", "yellow","tan", "pink", "purple", "gray", "white", "339900","666699","996699","black"
#)

# Plot 

s <- as.factor(wrc_family$Sample)
wrc_family$Sample <- ordered(s,levels=c("bulkM","avM", "ecM","bulkMF","avMF", "ecMF"))

wrc_plot<- ggplot(wrc_order, aes(x = Order, y = Abundance, fill =Sample)) + 
  #facet_grid(Type ~ .) +
  geom_bar(stat = "identity", width=0.5) +
  scale_fill_manual(values = family_colors) +
  # Remove x axis title
  theme(axis.title.x = element_blank()) + 
  guides(fill = guide_legend(keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Family > 1%) \n") +
  ggtitle("Order Composition of Bacterial Communities By Site") 

wrc_plot + coord_flip() 

ggsave(file="../figures/order_stack.pdf", width=15, height=9, dpi=300)

```

```{r phyloseq Family stacked graphs of taxa composition}
# melt to long format (for ggploting) 
# prune out phyla below 1% in each sample
wrc_family <- treat_merge %>%
  tax_glom(taxrank = "Family") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.01, Class == "Actinobacteria") %>%                        # Filter out low abundance taxa
  arrange(Family)                                     # Sort data frame alphabetically by phylum

# wrc_family
# Set colors for plotting
family_colors <- c("brown","orange", "blue", "red","yellow","purple")
  #"#CBD588", "#5F7FC7","#DA5724", "#508578", "#CD9BCD",
  # "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
 #"#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861", "#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7", "green", "red", "yellow","tan", "pink", "purple", "gray", "white", "339900","666699","996699","black"
#)

# Plot 

s <- as.factor(wrc_family$Sample)
t <- revalue(s, c("bulkM"="bulk unfertilized", "avM"="grass unfertilized", "ecM"="forb unfertilized","bulkMF"="bulk fertilized", "avMF"="grass fertilized", "ecMF"="forb fertilized"))
t <- mapvalues(s, from = c("bulkM","avM", "ecM","bulkMF","avMF", "ecMF"), to = c("bulk unfertilized","grass unfertilized", "forb unfertilized","bulk fertilized", "grass fertilized", "forb fertilized"))
t<-ordered(t,levels=c("bulk unfertilized","grass unfertilized", "forb unfertilized","bulk fertilized", "grass fertilized", "forb fertilized"))

wrc_plot<- ggplot(wrc_family, aes(x = Family, y = Abundance, fill = t)) + 
  #facet_grid(Type ~ .) +
  geom_bar(stat = "identity", width=0.5) +
  scale_fill_manual(values = family_colors) +
  # Remove x axis title
  theme_bw()  +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
          axis.title.y=element_text(margin=margin(r=10)), 
          axis.title.x=element_text(margin=margin(t=10)), panel.border =
          element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + 
  guides(fill = guide_legend(keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance > 1%") +
  xlab("Family Name")+
  labs(fill="Treatment")+
  ggtitle("Family Composition of Bacterial Communities") 

wrc_plot + coord_flip() 

wrc_family_o<-wrc_family[,c(2:3,40:45)]

write.csv(wrc_family_o, file="family_composition.csv", row.names=TRUE)

ggsave(file="../figures/family_stack.pdf", width=15, height=9, dpi=300)

```

```{r phyloseq Genus stacked graphs of taxa composition}
# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample
wrc_family <- treat_merge %>%
  tax_glom(taxrank = "Genus") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.01) %>%                        # Filter out low abundance taxa
  arrange(Genus)                                     # Sort data frame alphabetically by phylum

 wrc_family
# Set colors for plotting
family_colors <- c(
  "#CBD588", "#5F7FC7","#DA5724", "#508578", "#CD9BCD",
   "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
  "#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861", "#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7", "green", "red", "yellow","tan", "pink", "purple", "gray", "white", "339900","666699","black"
)

# Plot 

s <- as.factor(wrc_family$Sample)
wrc_family$Sample <- ordered(s,levels=c("bulkM","avM", "ecM","bulkMF","avMF", "ecMF"))

wrc_plot<- ggplot(wrc_family, aes(x = Genus, y = Abundance, fill = Sample)) + 
  #facet_grid(Type ~ .) +
  geom_bar(stat = "identity", width=0.5) +
  scale_fill_manual(values = family_colors) +
  # Remove x axis title
  theme(axis.title.x = element_blank()) + 
  guides(fill = guide_legend(keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Genus > 1%) \n") +
  ggtitle("Genus Composition of Bacterial Communities By Site") 

wrc_plot + coord_flip() 
ggsave(file="../figures/genus_stack.1.pdf", width=15, height=9, dpi=300)
```


```{r Bacteria Indicator Species}
#This produces a text file with otus for groups and p values
new.data <-cbind(meta,otu.2.rel)
library("labdsv")

design.type <- new.data$pxt
dataREL.2013 <- new.data[,-c(1:28)]
head(dataREL.2013)
set.seed(42)
dataREL <- dataREL.2013[, colSums(dataREL.2013) > 0.05]
bac.ind <- indval(dataREL, clustering =  design.type, numitr=1000, set.seed(42))
levels(design.type)
design.type
summary(bac.ind)

inds <- which(bac.ind$pval <= 0.05)
bac.indicators <- as.data.frame(matrix(NA, nrow = length(inds), ncol = 4))
colnames(bac.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")

bac.indicators$OTU <- names(inds)
bac.indicators$Cluster <- bac.ind$maxcls[inds]
bac.indicators$IndVal <- bac.ind$indcls[inds]
bac.indicators$Prob <- bac.ind$pval[inds]

otu.tax<-as.data.frame(tax)
suffix<- 1:nrow(otu.tax)
ps <- sprintf("Otu%05d",suffix)
otu.tax$OTU<-ps

ind.tax <- otu.tax[which(as.character(otu.tax$OTU) %in% bac.indicators$OTU), ]
ind.tax <- ind.tax[match(ind.tax$OTU, bac.indicators$OTU), ]

indicator.bac <- cbind(bac.indicators, ind.tax[, -c(1)])

indicator.bac <- indicator.bac[order(as.numeric(indicator.bac$Cluster)), ]

table(indicator.bac$Cluster)
table(indicator.bac$Phylum)
table(indicator.bac$Cluster)
levels(design.type)

# Export Bacteria Indicator Table
write.table(indicator.bac, "BacterialIndicators_Plant.txt",
            sep="\t", row.names = F, quote = F)

#From what I can tell by comparing both methods of indicator analysis the second "cluster" 1, 2, 3, ... is the design.type in alphabetical order. So 1 = avM, 2 = avMF, 3 = bulkM, 4 = bulkMF, 5 = ecM, 6=ecMF
  
```

```{r indicator species bubble plot}
require("stringr")

otu.2.meta <- as.data.frame(cbind(otu.2, meta))
rownames(otu.2.meta)
meta$group
otu.2.meta$group <- meta$pxt

d <- select_(otu.2.meta, "Otu00035","Otu00049","Otu00122", "Otu00089", "Otu00092" ,"Otu00129", "Otu00039" ,"Otu00063", "Otu00064", "Otu00104", "Otu00016","Otu00057", "Otu00094" ,"Otu00004", "Otu00024", "Otu00034", "Otu00076","Otu00008", "Otu00021" ,"Otu00027", "Otu00071", "group") 

d <-aggregate(d, by=list(d$group), FUN=mean, na.rm=TRUE)
rownames(d) = d$Group.1 
d <- d[,-c(1,23)]

d.rel <- d
for(i in 1:dim(d)[1]){
  d.rel[i,] <- d[i,]/sum(d[i,])
}

d.rel$group <- rownames(d)
d.rel <- melt(d.rel)

pcoa.shape <- d.rel$group
pcoa.shape <- ordered(pcoa.shape,levels=c("bulkM","bulkMF", "avM","avMF","ecM","ecMF"))

p1 <- ggplot(d.rel, aes(x=variable, y=pcoa.shape, fill=pcoa.shape)) #, fill=value)) 
p1+geom_point(aes(size=value), shape=21)+
 scale_fill_manual(values=c("#EB8F43", "#9BBB59","#EB8F43", "#9BBB59","#EB8F43", "#9BBB59","#EB8F43", "#9BBB59"))+ guides(fill=F)+
  scale_y_discrete(labels=c("Bulk","Bulk","Grass","Grass","Forb","Forb"))+
 #scale_fill_gradientn(colors=rainbow(5),limits=c(0,.5), guide_legend("Relative Abundance"))+
  scale_x_discrete(labels=c("35","49","122","89","92","129","39","63","64","104","16","57","94","4","24", "34","76","8","21","27","71")) +
theme(panel.background=element_blank(), panel.border = element_rect(colour = "black", fill=NA, size=1)) + scale_size_area(max_size=15, limits=c(0,0.5)) + theme(axis.text.x=element_text(size=12, angle=90),
          axis.text.y=element_text(size=14, angle=0), legend.text=element_text(size=14), axis.title = (element_text(size=18)), axis.ticks.length=unit(.25, "cm")) +
labs(x="OTU", y="Soil Source") + labs(size=expression("Relative Abundance"))


```

```{r indicator species analysis using multipatt}
#Value (IndVal) index to measure the association between a species and a
#site group. The method of Dufr^ene and Legendre [1997] calculates the In-
#dVal index between the species and each site group and then looks for the
#group corresponding to the highest association value.
require("indicspecies")

groups= metaotu$pxt
groups
#block = metaotu$Block
#block
m <- as.data.frame(otu.2.rel[, colSums(otu.2.rel) > 0.05], na.rm=T)
#m <- m[,1:1000]
set.seed(42)
#is <- multipatt(m, groups, restcomb=c(1,2,3,4,5,6,27,36), control=how(nperm=1000), 
#is <- multipatt(m, groups, restcomb=c(27,36), control=how(nperm=999), set.seed(42))
is <- multipatt(m, groups, restcomb=c(1,2,3,4,5,6), control=how(nperm=1000), func="IndVal.g",duleg=T, set.seed(42))

#is_block <-multipatt(otu.2.rel, block, control=how(nperm=1000))

#c<-coverage(m, is)

summary(is)
#summary(is_block)

is$sign
```

```{r species comparison}
#Work in progress
#Compare list of OTUs in grass m/mf and forb m/mf
#Core micorbiome within plant species, among plant species?

#Combine otu list and meta data so can filter otus
metaotu.2 <- cbind(otu.2, meta)
#head(metaotu.2)
#tail(metaotu.2)

#Get Otu list for grass M
metaotu.2.avM <- metaotu.2[which(metaotu.2$pxt=='avM'),]
#Remove meta data, checking where meta data begins/ends
metaotu.2.avM.t<-metaotu.2.avM[,c(10029:10063)]
metaotu.2.avM<-metaotu.2.avM[,-c(10029:10063)]
#Tranpose data frame
metaotu.2.avM.t <- as.data.frame(t(metaotu.2.avM))
head(metaotu.2.avM.t)
#Select only the OTUs that appear in every field replicate of a treatment 
colnames(metaotu.2.avM.t)
#select rows that are not equal to zero, all columns
metaotu.2.avM.t <- metaotu.2.avM.t[which(metaotu.2.avM.t$WRC15AVRS2M!=0 & metaotu.2.avM.t$WRC15AVRS4M!=0 &  metaotu.2.avM.t$WRC15AVRS6M!=0 & metaotu.2.avM.t$WRC15AVRS8M!=0),]
avM.otu <- rownames(metaotu.2.avM.t)

#Get Otu list for grass MF
metaotu.2.avMF <- metaotu.2[which(metaotu.2$pxt=='avMF'),]
#Remove meta data, checking where meta data begina/ends
metaotu.2.avMF.t<-metaotu.2.avMF[,c(10029:10063)]
metaotu.2.avMF<-metaotu.2.avMF[,-c(10029:10063)]
metaotu.2.avMF.t <- as.data.frame(t(metaotu.2.avMF))
head(metaotu.2.avMF.t)
#Select only the OTUs that appear in every treatment of one type
colnames(metaotu.2.avMF.t)
metaotu.2.avMF.t <- metaotu.2.avMF.t[which((metaotu.2.avMF.t$WRC15AVRS2MF!=0 & metaotu.2.avMF.t$WRC15AVRS4MF!=0 & metaotu.2.avMF.t$WRC15AVRS6MF!=0 & metaotu.2.avMF.t$WRC15AVRS8MF!=0)),]
avMF.otu <- rownames(metaotu.2.avMF.t)
#Which taxa are the same between avm and avmf?
av.diff <- intersect(avM.otu, avMF.otu)
tax <- as.data.frame(tax_table(meta_merge))
otu = row.names(tax)
tax <- cbind(otu, tax)
tax.av.core <- tax %>% filter(row.names(tax) %in% av.diff)

#Sum occurences of a particular rank
tax.av.core.r <- tax.av.core %>% 
  group_by(Rank2) %>%
  tally()
#Add field for the percentage of the total for taxa in that rank
col_sum <- sum(tax.av.core.r$n)
av.p <- within(tax.av.core.r, p <- (n/col_sum)*100)
#add field to denote these are from the grass samples
group = "g"
tax.av.core.r <- cbind(group, tax.av.core.r)
av.p <- cbind(group, av.p)

av.core.plot <- ggplot(tax.av.core.r, aes(x=Rank2, y=n)) + geom_point() + coord_flip()
av.core.plot

#Get Otu list for forb M
metaotu.2.ecM <- metaotu.2[which(metaotu.2$pxt=='ecM'),]
#Remove meta data, checking where meta data begins/ends
metaotu.2.ecM.t<-metaotu.2.ecM[,c(18206:18224)]
metaotu.2.ecM<-metaotu.2.ecM[,-c(18206:18224)]
#Tranpose data frame
metaotu.2.ecM.t <- as.data.frame(t(metaotu.2.ecM))
head(metaotu.2.ecM.t)
#Select only the OTUs that appear in every field replicate of a treatment 
colnames(metaotu.2.ecM.t)
#select rows that are not equal to zero, all columns
metaotu.2.ecM.t <- metaotu.2.ecM.t[which(metaotu.2.ecM.t$WRC15ECRS2M!=0 & metaotu.2.ecM.t$WRC15ECRS4M!=0 &  metaotu.2.ecM.t$WRC15ECRS6M!=0 & metaotu.2.ecM.t$WRC15ECRS8M!=0),]
ecM.otu <- rownames(metaotu.2.ecM.t)

#Get Otu list for forb MF
metaotu.2.ecMF <- metaotu.2[which(metaotu.2$pxt=='ecMF'),]
#Remove meta data, checking where meta data begina/ends
metaotu.2.ecMF.t<-metaotu.2.ecMF[,c(18206:18224)]
metaotu.2.ecMF<-metaotu.2.ecMF[,-c(18206:18224)]
metaotu.2.ecMF.t <- as.data.frame(t(metaotu.2.ecMF))
head(metaotu.2.ecMF.t)
#Select only the OTUs that appear in every treatment of one type
colnames(metaotu.2.ecMF.t)
metaotu.2.ecMF.t <- metaotu.2.ecMF.t[which((metaotu.2.ecMF.t$WRC15ECRS2MF!=0 & metaotu.2.ecMF.t$WRC15ECRS4MF!=0 & metaotu.2.ecMF.t$WRC15ECRS6MF!=0 & metaotu.2.ecMF.t$WRC15ECRS8MF!=0)),]
ecMF.otu <- rownames(metaotu.2.ecMF.t)

#Otus that are the same
ec.diff <- intersect(ecM.otu, ecMF.otu)
ec.diff
#Join with taxonomy file
tax <- as.data.frame(tax_table(meta_merge))
otu = row.names(tax)
tax <- cbind(otu, tax)
tax.ec.core <- tax %>% filter(row.names(tax) %in% ec.diff)

#Count occurences of taxa in a rank
tax.ec.core.r <- tax.ec.core %>% 
  group_by(Rank2) %>%
  tally()
#calculate the percentage of the total that each taxa represents
col_sum <- sum(tax.ec.core.r$n)
ec.p <- within(tax.ec.core.r, p <- (n/col_sum)*100)
#add column to denote from forb gorup
group = "f"
tax.ec.core.r <- cbind(group, tax.ec.core.r)
ec.p <- cbind(group, ec.p)

ec.core.plot <- ggplot(tax.ec.core.r, aes(x=Rank2, y=n)) + geom_point() + coord_flip()
ec.core.plot

#which otus are the same between grass and forb?
ec.av.diff <- intersect(ec.diff, av.diff)
ec.av.diff
#which otus are different in forb compared ot grass?
ecdiff <- setdiff(ec.diff, av.diff)
ecdiff

tax <- as.data.frame(tax_table(meta_merge))
otu = row.names(tax)
tax <- cbind(otu, tax)
tax.ecdiff <- tax %>% filter(row.names(tax) %in% ecdiff)

tax.ecdiff.r <- tax.ecdiff %>% 
  group_by(Rank2) %>%
  tally()

#Whch otus are diff in grass compared to forb?
avdiff <- setdiff(av.diff, ec.diff)
avdiff

tax <- as.data.frame(tax_table(meta_merge))
otu = row.names(tax)
tax <- cbind(otu, tax)
tax.avdiff <- tax %>% filter(row.names(tax) %in% avdiff)

tax.avdiff.r <- tax.avdiff %>% 
  group_by(Rank2) %>%
  tally()

#assign taxonomy to otus for otus similar in grass and forb
tax <- as.data.frame(tax_table(meta_merge))
otu = row.names(tax)
tax <- cbind(otu, tax)
tax.ec.av.core <- tax %>% filter(row.names(tax) %in% ec.av.diff)

tax.ec.av.core.r <- tax.ec.av.core %>% 
  group_by(Rank2) %>%
  tally()

col_sum <- sum(tax.ec.av.core.r$n)
ec.av.p <- within(tax.ec.av.core.r, p <- (n/col_sum)*100)

#add fg to denote these are similar in grass and forb
group = "fg"
tax.ec.av.core.r <- cbind(group, tax.ec.av.core.r)
ec.av.p <- cbind(group, ec.av.p)

ec.av.core.plot <- ggplot(tax.ec.av.core.r, aes(x=Rank2, y=n)) + geom_point() + coord_flip()
ec.av.core.plot

#combine dataframes so can plot together
combo <- rbind(tax.ec.av.core.r, tax.ec.core.r, tax.av.core.r)

combop <- rbind(ec.p, av.p, ec.av.p)

p <- ggplot() + geom_bar(aes(y=p, x=group, fill=Rank2), data=combop,stat="identity")# + coord_flip()
p                           

```

```{r species comparison - Rank 4}

#Compare list of OTUs in grass m/mf and forb m/mf
#Core micorbiome within plant species, among plant species?

#Combine otu list and meta data so can filter otus
metaotu.2 <- cbind(otu.2, meta)
#head(metaotu.2)
#tail(metaotu.2)

#Get Otu list for grass M
metaotu.2.avM <- metaotu.2[which(metaotu.2$pxt=='avM'),]
#Remove meta data, checking where meta data begins/ends
metaotu.2.avM.t<-metaotu.2.avM[,c(18206:18224)]
metaotu.2.avM<-metaotu.2.avM[,-c(18206:18224)]
#Tranpose data frame
metaotu.2.avM.t <- as.data.frame(t(metaotu.2.avM))
head(metaotu.2.avM.t)
#Select only the OTUs that appear in every field replicate of a treatment 
colnames(metaotu.2.avM.t)
#select rows that are not equal to zero, all columns
metaotu.2.avM.t <- metaotu.2.avM.t[which(metaotu.2.avM.t$WRC15AVRS2M!=0 & metaotu.2.avM.t$WRC15AVRS4M!=0 &  metaotu.2.avM.t$WRC15AVRS6M!=0 & metaotu.2.avM.t$WRC15AVRS8M!=0),]
avM.otu <- rownames(metaotu.2.avM.t)

#Get Otu list for grass MF
metaotu.2.avMF <- metaotu.2[which(metaotu.2$pxt=='avMF'),]
#Remove meta data, checking where meta data begina/ends
metaotu.2.avMF.t<-metaotu.2.avMF[,c(18206:18224)]
metaotu.2.avMF<-metaotu.2.avMF[,-c(18206:18224)]
metaotu.2.avMF.t <- as.data.frame(t(metaotu.2.avMF))
head(metaotu.2.avMF.t)
#Select only the OTUs that appear in every treatment of one type
colnames(metaotu.2.avMF.t)
metaotu.2.avMF.t <- metaotu.2.avMF.t[which((metaotu.2.avMF.t$WRC15AVRS2MF!=0 & metaotu.2.avMF.t$WRC15AVRS4MF!=0 & metaotu.2.avMF.t$WRC15AVRS6MF!=0 & metaotu.2.avMF.t$WRC15AVRS8MF!=0)),]
avMF.otu <- rownames(metaotu.2.avMF.t)
#Which taxa are the same between avm and avmf?
av.diff <- intersect(avM.otu, avMF.otu)
tax <- as.data.frame(tax_table(meta_merge))
otu = row.names(tax)
tax <- cbind(otu, tax)
tax.av.core <- tax %>% filter(row.names(tax) %in% av.diff)

#Sum occurences of a particular rank
tax.av.core.r <- tax.av.core %>% 
  group_by(Rank4) %>%
  tally()
#Add field for the percentage of the total for taxa in that rank
col_sum <- sum(tax.av.core.r$n)
av.p <- within(tax.av.core.r, p <- (n/col_sum)*100)
#add field to denote these are from the grass samples
group = "g"
tax.av.core.r <- cbind(group, tax.av.core.r)
av.p <- cbind(group, av.p)

av.core.plot <- ggplot(tax.av.core.r, aes(x=Rank4, y=n)) + geom_point() + coord_flip()
av.core.plot

#Get Otu list for forb M
metaotu.2.ecM <- metaotu.2[which(metaotu.2$pxt=='ecM'),]
#Remove meta data, checking where meta data begins/ends
metaotu.2.ecM.t<-metaotu.2.ecM[,c(18206:18224)]
metaotu.2.ecM<-metaotu.2.ecM[,-c(18206:18224)]
#Tranpose data frame
metaotu.2.ecM.t <- as.data.frame(t(metaotu.2.ecM))
head(metaotu.2.ecM.t)
#Select only the OTUs that appear in every field replicate of a treatment 
colnames(metaotu.2.ecM.t)
#select rows that are not equal to zero, all columns
metaotu.2.ecM.t <- metaotu.2.ecM.t[which(metaotu.2.ecM.t$WRC15ECRS2M!=0 & metaotu.2.ecM.t$WRC15ECRS4M!=0 &  metaotu.2.ecM.t$WRC15ECRS6M!=0 & metaotu.2.ecM.t$WRC15ECRS8M!=0),]
ecM.otu <- rownames(metaotu.2.ecM.t)

#Get Otu list for forb MF
metaotu.2.ecMF <- metaotu.2[which(metaotu.2$pxt=='ecMF'),]
#Remove meta data, checking where meta data begina/ends
metaotu.2.ecMF.t<-metaotu.2.ecMF[,c(18206:18224)]
metaotu.2.ecMF<-metaotu.2.ecMF[,-c(18206:18224)]
metaotu.2.ecMF.t <- as.data.frame(t(metaotu.2.ecMF))
head(metaotu.2.ecMF.t)
#Select only the OTUs that appear in every treatment of one type
colnames(metaotu.2.ecMF.t)
metaotu.2.ecMF.t <- metaotu.2.ecMF.t[which((metaotu.2.ecMF.t$WRC15ECRS2MF!=0 & metaotu.2.ecMF.t$WRC15ECRS4MF!=0 & metaotu.2.ecMF.t$WRC15ECRS6MF!=0 & metaotu.2.ecMF.t$WRC15ECRS8MF!=0)),]
ecMF.otu <- rownames(metaotu.2.ecMF.t)

#Otus that are the same
ec.diff <- intersect(ecM.otu, ecMF.otu)
ec.diff
#Join with taxonomy file
tax <- as.data.frame(tax_table(meta_merge))
otu = row.names(tax)
tax <- cbind(otu, tax)
tax.ec.core <- tax %>% filter(row.names(tax) %in% ec.diff)

#Count occurences of taxa in a rank
tax.ec.core.r <- tax.ec.core %>% 
  group_by(Rank4) %>%
  tally()
#calculate the percentage of the total that each taxa represents
col_sum <- sum(tax.ec.core.r$n)
ec.p <- within(tax.ec.core.r, p <- (n/col_sum)*100)
#add column to denote from forb gorup
group = "f"
tax.ec.core.r <- cbind(group, tax.ec.core.r)
ec.p <- cbind(group, ec.p)

ec.core.plot <- ggplot(tax.ec.core.r, aes(x=Rank4, y=n)) + geom_point() + coord_flip()
ec.core.plot

#which otus are the same between grass and forb?
ec.av.diff <- intersect(ec.diff, av.diff)
ec.av.diff
#which otus are different in forb compared ot grass?
ecdiff <- setdiff(ec.diff, av.diff)
ecdiff

tax <- as.data.frame(tax_table(meta_merge))
otu = row.names(tax)
tax <- cbind(otu, tax)
tax.ecdiff <- tax %>% filter(row.names(tax) %in% ecdiff)

tax.ecdiff.r <- tax.ecdiff %>% 
  group_by(Rank4) %>%
  tally()

#Whch otus are diff in grass compared to forb?
avdiff <- setdiff(av.diff, ec.diff)
avdiff

tax <- as.data.frame(tax_table(meta_merge))
otu = row.names(tax)
tax <- cbind(otu, tax)
tax.avdiff <- tax %>% filter(row.names(tax) %in% avdiff)

tax.avdiff.r <- tax.avdiff %>% 
  group_by(Rank4) %>%
  tally()

#assign taxonomy to otus for otus similar in grass and forb
tax <- as.data.frame(tax_table(meta_merge))
otu = row.names(tax)
tax <- cbind(otu, tax)
tax.ec.av.core <- tax %>% filter(row.names(tax) %in% ec.av.diff)

tax.ec.av.core.r <- tax.ec.av.core %>% 
  group_by(Rank4) %>%
  tally()

col_sum <- sum(tax.ec.av.core.r$n)
ec.av.p <- within(tax.ec.av.core.r, p <- (n/col_sum)*100)

#add fg to denote these are similar in grass and forb
group = "fg"
tax.ec.av.core.r <- cbind(group, tax.ec.av.core.r)
ec.av.p <- cbind(group, ec.av.p)

ec.av.core.plot <- ggplot(tax.ec.av.core.r, aes(x=Rank4, y=n)) + geom_point() + coord_flip()
ec.av.core.plot

#combine dataframes so can plot together
combo <- rbind(tax.ec.av.core.r, tax.ec.core.r, tax.av.core.r)

combop <- rbind(ec.p, av.p, ec.av.p)

p <- ggplot() + geom_bar(aes(y=p, x=group, fill=Rank4), data=combop,stat="identity")# + coord_flip()
p                           

```


